app/services/import_polygons.rb

```diff
@@ -114,4 +103,22 @@ def buffer_api_endpoint(coords, distance)
   def take_every_nth_coord(coords, number)
     coords.each_with_index.filter_map { |item, index| item if (index % number).zero? }
   end
+
+  def poylgons_have_changed?(location_polygon, polygons)
+    return true if location_polygon&.buffers.nil? || location_polygon.buffers&.fetch(BUFFER_RADII.first.to_s, nil).nil?
+
+    get_buffered_polygons(polygons, BUFFER_RADII.first) != location_polygon.buffers[BUFFER_RADII.first.to_s]
+  end
+
+  def get_buffered_polygons(polygons, distance)
```

Comment:
You're now calling this method twice per location for the first radius option (once for `poylgons_have_changed?` and once in `buffered_polygons`), which will make the task take longer (I assume it's getting the buffered polygons that takes so much time).

However, I can't immediately think of a good way to avoid this. I can only think of
- _assign_ the output of `get_buffered_polygons` for the first radius option to a variable
- check it against the buffer in the database
- when you update the location polygon, update the first radius option using the variable you already assigned, and then make calls for the remaining radius options.

This seems a bit convoluted.

As the task only runs once per week and isn't causing issues at the moment, I'm happy to leave this duplication alone rather than make the code more complicated and hard to follow.